src/nyaml/nxdl2nyaml.py

```diff
@@ -777,34 +793,42 @@ def handle_dim_with_all_dim_attr(depth, node, file_out, possible_dim_attrs):
-                    file_out.write(f"{indent}{attr}: {value}\n")
-                else:
-                    raise ValueError(
-                        f"Dimension has an attribute {attr} that is not valid."
-                        f"Currently allowed attributes are {possible_dimension_attrs}."
-                        f"Please have a close look"
-                    )
-            # Taking care of dimension doc
-            dimension_doc = ""
-            for child in list(node):
-                tag = remove_namespace_from_tag(child.tag)
-                if tag == "doc":
-                    dimension_doc = self.handle_not_root_level_doc(
-                        depth + 1, child.text
-                    )
-                    file_out.write(dimension_doc)
-                    node.remove(child)
-            handle_dim_with_all_dim_attr(depth, node, file_out, possible_dim_attrs)
-        # Dimension has only index and value
+        if len(node) > 0:
+            if remove_namespace_from_tag(node[0].tag) == "doc" or len(node.attrib) > 0:
+                indent = depth * DEPTH_SIZE
+                tag = remove_namespace_from_tag(node.tag)
+                file_out.write(f"{indent}{tag}:\n")
+                for attr, value in node.attrib.items():
+                    if attr in possible_dimension_attrs and not isinstance(value, dict):
+                        indent = (depth + 1) * DEPTH_SIZE
+                        file_out.write(f"{indent}{attr}: {value}\n")
+                    else:
+                        raise ValueError(
+                            f"Dimension has an attribute {attr} that is not valid."
+                            f"Currently allowed attributes are {possible_dimension_attrs}."
+                            f"Please have a close look"
+                        )
+                # Taking care of dimension doc
+                dimension_doc = ""
+                for child in list(node):
+                    tag = remove_namespace_from_tag(child.tag)
+                    if tag == "doc":
+                        dimension_doc = self.handle_not_root_level_doc(
+                            depth + 1, child.text
+                        )
+                        file_out.write(dimension_doc)
+                        node.remove(child)
+                handle_dim_with_all_dim_attr(depth, node, file_out, possible_dim_attrs)
+            # Dimension has only index and value
+            else:
+                handle_dim_with_value_and_index(depth, node, file_out, dimension_doc="")
         else:
-            handle_dim_with_value_and_index(depth, node, file_out, dimension_doc="")
+            # dimension has only rank as seen in NXfresnel_zone_plate.nxdl.xml
+            if len(node.attrib) == 1 and "rank" in node.attrib:
+                indent = depth * DEPTH_SIZE
+                file_out.write(f"{indent}dim: (,)\n")
+            else:
+                raise NotImplementedError()
```

Comment:
Should we allow someone to define a `dimension` without proper documentation
This is not needed. The `prossible_dimension_attrs` contains the `rank` for dimension node.

Reply:
Nevertheless the case that a node.attrib dict has only one key, value pair namely rank was not catched so far that's why the code modification is substantiated

Reply:
I think you are refering https://github.com/FAIRmat-NFDI/nexus_definitions/blob/main/base_classes/NXfresnel_zone_plate.nxdl.xml#L37 , which is not correct. Should we allow any faulty file to parse? We are also using `nyaml` for us. We want the tool not to allow to parse any faulty file, which enhances our development experience of app def. After scrunity review of our app def, very often we get issue in app def. I do not want to make that worse even, only for handling error files. 

We must propose NAIC to review that files which is sustainable.

Reply:
Here I see that `handle_dim_with_value_and_index` is removed`, where and how `index` and `value` attr pairs are being handled?